tests/js/spec/utils/withCommitters.spec.jsx

```diff
@@ -0,0 +1,161 @@
+import React from 'react';
+
+import {mount} from 'sentry-test/enzyme';
+
+import CommitterStore from 'app/stores/committerStore';
+import withCommitters from 'app/utils/withCommitters';
+
+describe('withCommitters HoC', function() {
+  const organization = TestStubs.Organization();
+  const project = TestStubs.Project();
+  const event = TestStubs.Event();
+  const group = TestStubs.Group({firstRelease: {}});
+
+  const endpoint = `/projects/${organization.slug}/${project.slug}/events/${event.id}/committers/`;
+
+  const api = new MockApiClient();
+  const mockData = {
+    committers: [
+      {
+        author: TestStubs.CommitAuthor(),
+        commits: [TestStubs.Commit()],
+      },
+    ],
+  };
+
+  beforeEach(() => {
+    MockApiClient.clearMockResponses();
+    MockApiClient.addMockResponse({
+      url: endpoint,
+      body: mockData,
+    });
+
+    jest.restoreAllMocks();
+    CommitterStore.init();
+  });
+
+  it('adds committers prop', async () => {
+    const Component = () => null;
+    const Container = withCommitters(Component);
+    const wrapper = mount(
+      <Container
+        api={api}
+        organization={organization}
+        project={project}
+        event={event}
+        group={group}
+      />
+    );
+
+    await tick(); // Run Store.load
+    await tick(); // Run Store.loadSuccess
+    wrapper.update(); // Re-render component with Store data
+
+    const mountedComponent = wrapper.find(Component);
+    expect(mountedComponent.prop('committers')).toEqual(mockData.committers);
+    expect(mountedComponent.prop('committersLoading')).toEqual(undefined);
+    expect(mountedComponent.prop('committersError')).toEqual(undefined);
+  });
+
+  it('prevents repeated calls', async () => {
+    const Component = () => null;
+    const Container = withCommitters(Component);
+
+    // XXX(leedongwei): We cannot spy on `fetchCommitters` as Jest can't
+    // replace the method in the prototype due to createReactClass.
+    // As such, I'm using `componentDidMount` as a proxy.
+    jest.spyOn(api, 'requestPromise');
+    jest.spyOn(Container.prototype, 'componentDidMount');
+    // jest.spyOn(Container.prototype, 'fetchCommitters');
+
+    // Mount and run component
+    mount(
+      <Container
+        api={api}
+        organization={organization}
+        project={project}
+        event={event}
+        group={group}
+      />
+    );
+    await tick();
+    await tick();
+
+    // Mount and run duplicates
+    mount(
+      <Container
+        api={api}
+        organization={organization}
+        project={project}
+        event={event}
+        group={group}
+      />
+    );
+    await tick();
+    mount(
+      <Container
```

Comment:
instead of remounting, maybe you could the jest method `.update()` ?

Reply:
This isn't remounting - I'm trying to simulate having several instances components on a single page.

Reply:
Would that be clearer if the components were nested or as siblings to a common parent?

Reply:
I have 2 tests for duplicated components, where (1) three instances load at the same time, (2) three instances that load one after the other. 

In this test, it is (2), so I have to mount them one-by-one and cannot nest them together.